examples/lpt_nbody_demo.py

```diff
@@ -31,110 +30,89 @@ def shmap(f: Callable,
   """Helper function to create a shard_map function that extracts the mesh from the
     context."""
-  # Extracts the mesh from the context
   mesh = mesh_lib.thread_resources.env.physical_mesh
   return shard_map(f, mesh, in_specs, out_specs, check_rep, auto)
 
 
-def _global_to_local_size(mesh_shape):
-  """ Utility function to compute the expected local size of a mesh
-      given the global size and the sharding strategy.
+def _global_to_local_size(nc: int):
+  """ Helper function to get the local size of a mesh given the global size.
   """
   pdims = mesh_lib.thread_resources.env.physical_mesh.devices.shape
-  return [mesh_shape[0] // pdims[1], mesh_shape[1] // pdims[0], mesh_shape[2]]
+  return [nc // pdims[0], nc // pdims[1], nc]
 
 
 def fttk(nc: int) -> list:
   """
     Generate Fourier transform wave numbers for a given mesh.
 
-    Parameters
-    ----------
-    mesh_shape : int
-        Shape of the mesh grid.
-    sharding : Any
-        Sharding strategy for the array.
-
-    Returns
-    -------
-    list
-        List of wave number arrays for each dimension.
+    Args:
+        nc (int): Shape of the mesh grid.
+
+    Returns:
+        list: List of wave number arrays for each dimension.
   """
   kd = np.fft.fftfreq(nc) * 2 * np.pi
 
   @partial(
       shmap,
-      in_specs=(P('z'), P('y'), P(None)),
-      out_specs=(P('z'), P(None, 'y'), P(None)))
+      in_specs=(P('x'), P('y'), P(None)),
+      out_specs=(P('x'), P(None, 'y'), P(None)))
   def get_kvec(kx, ky, kz):
-    return (kx.reshape([-1, 1, 1]), ky.reshape([1, -1,
-                                                1]), kz.reshape([1, 1, -1]))
+    return (kx.reshape([-1, 1, 1]),
+            ky.reshape([1, -1, 1]),
+            kz.reshape([1, 1, -1])) # yapf: disable
 
   return get_kvec(kd, kd, kd)
 
 
 def gravity_kernel(kvec):
-  """ Fourier kernel to compute gravitational forces from a Fourier space density field.
+  """ Computes a Fourier kernel combining laplace and derivative
+    operators to compute gravitational forces.
 
-    Parameters
-    ----------
-    kvec : tuple of float
-        Wave vector in Fourier space.
+    Args:
+        kvec (tuple of float): Wave numbers in Fourier space.
 
-    Returns
-    -------
-    jnp.ndarray
-        Gravitational kernel.
+    Returns:
+        tuple of jnp.ndarray: kernels for each dimension.
   """
   kx, ky, kz = kvec
   kk = kx**2 + ky**2 + kz**2
   laplace_kernel = jnp.where(kk == 0, 1., 1. / kk)
-  grav_kernel = [
-      laplace_kernel * 1j / 6.0 * (8 * jnp.sin(kx) - jnp.sin(2 * kx)),
-      laplace_kernel * 1j / 6.0 * (8 * jnp.sin(ky) - jnp.sin(2 * ky)),
-      laplace_kernel * 1j / 6.0 * (8 * jnp.sin(kz) - jnp.sin(2 * kz))
-  ]
+  # Note that we return frequency arrays in the transposed order [z, x, y]
+  # corresponding to the transposed FFT output
+  grav_kernel = (laplace_kernel * 1j * kz,
+                 laplace_kernel * 1j * kx,
+                 laplace_kernel * 1j * ky) # yapf: disable
```

Comment:
It would be simpler to do and easier to read

`kz ,kx, ky = kvec`

and use the more intuitive ordering in the gradient kernel